src/components/ProfileView.jsx

```diff
@@ -323,11 +359,12 @@ const MailConfirmationCode = translate()(({
       </Button>
       <Button
         onClick={() => checkMailConfirmationCode(fields.mail_confirmation_code.value)}
+        disabled={fields.email.value === ''}
```

Comment:
Same question here. As `!''` is `true`, we could only check for falsy/truthy value by just checking `fields.email.value`. It could also solve cases when `fields.email.value` is `null` or `undefined`.